src/libraries/System.Private.CoreLib/src/System/Collections/Generic/ArraySortHelper.cs

```diff
@@ -779,7 +767,7 @@ private static void DownHeap(Span<TKey> keys, Span<TValue> values, int i, int n,
             TKey d = keys[lo + i - 1];
             TValue dValue = values[lo + i - 1];
 
-            while (i <= n / 2)
+            while (i <= n >> 1)
```

Comment:
I'm just curious about this, is the JIT not able to transform `int / c`  where c is a power of 2 into `int >> Log2(c)`?

Reply:
`uint`, yes.  `int`, no.

Reply:
(Sorry if I've just asked the same question as what Tanner asked above, my thought was that the sign doesn't matter for divisions / etc)

Reply:
https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIGYACMhgYQYG8aHunGBLAHYYGAET4A3UgApBw7AEoGAXgB8DbAwD0DUgG4aAXyA===

It translates it to `((i >> 31) + i) >> 1)`, which is correct for signed types. presumably it can't rationalise the fact `i` is always positive here